pkg/e2e/upgrade_test.go

```diff
@@ -119,97 +118,108 @@ func Test_Upgrade(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// wait for our echo server to become ready
 	err = utils.WaitForPodReady(e2eSuite.kubeclient,
 		"echoserver", namespaceUpgradeTest)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	var execOut bytes.Buffer
-	var execErr bytes.Buffer
+	RESTClient, err := rest.RESTClientFor(restConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	// curl echo server from within pod
-	ioStreams := genericclioptions.IOStreams{In: nil, Out: &execOut, ErrOut: &execErr}
-	execOptions := &exec.ExecOptions{
-		StreamOptions: exec.StreamOptions{
-			IOStreams: ioStreams,
-			PodName:   pod.Name,
-			Namespace: pod.Namespace,
-		},
-		Command: []string{
-			"curl", "127.0.0.1:8080", "-s", "-d", "hello world",
-		},
-		PodClient: kubeclient.CoreV1(),
-		Config:    restConfig,
-		Executor:  &exec.DefaultRemoteExecutor{},
+	req := RESTClient.Post().
+		Resource("pods").
+		Name(pod.Name).
+		Namespace(pod.Namespace).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: "echoserver",
+			Command: []string{
+				"curl", "127.0.0.1:8080", "-s", "-d", "hello world",
+			},
+			Stdin:  false,
+			Stdout: true,
+			Stderr: true,
+			TTY:    false,
+		}, scheme.ParameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(restConfig, "POST", req.URL())
```

Comment:
Out of interest, does this mean a change in the supported k8s versions? What versions support SPDY?

Reply:
Only the k8s version we support :man_shrugging: 
1.11->15